### ISSUE TITLE
Issues with debundler approach for using irbtools in Rails

### ISSUE THREAD

**Issue by tovodeverett** [janlelis/irbtools#26]
I ran into an issue with using the debundle hack approach document in the Readme for using irbtools in Rails.  It looks like I'm not the only one, as I located https://github.com/rails/rails/issues/9684#issuecomment-14830915 which has the exact same symptoms I was running into.

I decided that when I'm running Bundler I want Bundler re-enabled after the stuff loads from my `~/.irbrc` to ensure that dynamic loads from with Rails do what they would normally do.  After a lot of trial and error, I came up with the following snippets.  Note that I do not have a comprehensive understanding of Ruby internals, so I don't claim to have the best solution, but it does appear to be working for me.

Snippet for using at the top of the `~/.irbrc` file:

```
# modified from https://github.com/ConradIrwin/pry-debundle/pull/5
if defined?(Gem.post_reset_hooks)
  temp_debundle_original_post_reset_hooks = Gem.post_reset_hooks.dup
  Gem.post_reset_hooks.reject!{ |hook| hook.source_location.first =~ %r{/bundler/} }
  Gem::Specification.reset
  alias temp_debundle_original_require require
  alias temp_debundle_original_gem gem
  load 'rubygems/custom_require.rb'
  alias gem require
end
```

Snippet for using at the very bottom of the `~/.irbrc` file:

```
if defined?(Gem.post_reset_hooks)
  Gem.post_reset_hooks.replace(temp_debundle_original_post_reset_hooks)
  Gem::Specification.reset
  alias require temp_debundle_original_require
  alias gem temp_debundle_original_gem
  Object.send :remove_method, :temp_debundle_original_require
  Object.send :remove_method, :temp_debundle_original_gem
end
```

Basically, I did my best to stash everything that got wiped out in the original debundle snippet so that I could restore it upon finishing up the load for the `~/.irbrc` file.
